### PR TITLE
chore: remove testLegacyBehavior and update testFutureBehavior tests

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
testLegacyBehavior only runs tests in v1 and skips them in v2, so none of these tests were running anyway. testFutureBehavior just runs the tests with the feature-flag as true, which is the default to these in v2 anyway. I'm removing these functions and their uses so that contributors aren't mislead to think they should use them.

I also fixed some spacing issues in test packages. The diff on this is kind of hard to follow but the summary of the change is this:
- all testLegacyBehavior tests have been deleted
- all testFutureBehavior tests have been updated to be standard tests
- describe blocks were removed where they wrapped around legacy or future sets of tests
- spacing fixed

----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudfront/test/distribution.test.ts [hunk 1]
```diff
@@ -475,61 +471,25 @@ describe('certificates', () => {
...
-          DistributionConfig: {
-            Aliases: ['example.com', 'www.example.com'],
-            ViewerCertificate: {
-              AcmCertificateArn: 'arn:aws:acm:us-east-1:123456789012:certificate/12345678-1234-1234-1234-123456789012',
-              SslSupportMethod: 'sni-only',
-              MinimumProtocolVersion: 'TLSv1.2_2019',
-            },
-          },
-        });
+  test('use the TLSv1.2_2021 security policy by default', () => {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-cloudfront/test/distribution.test.ts:474]: Don't we still need to test the legacy behavior though? For some of the feature flags it is possible to still set the flag to `false`.

Not sure if it would be better to keep `testLegacyBehavior` or just set the feature flag through `node.setContext()`

**Review by corymhall — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-cloudfront/test/distribution.test.ts (hunk 1 above)
Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/aws-cloudfront/test/distribution.test.ts:474]: There are places that the feature flags are set as false to do that very thing. testLegacyBehavior, however, just skips the tests in v2, it doesn't set the flags to false.

**Review by corymhall — APPROVED** [review]
